### PR TITLE
Reduce run_cypher calls for GDS endpoints

### PR DESCRIPTION
**Pull request by FlorentinD**
Thank you for your contribution to the Graph Data Science Client project.

Before submitting this PR, please read [Contributing to the Neo4j Ecosystem](https://github.com/neo4j/graph-data-science-client/blob/main/CONTRIBUTING.md). 

Make sure:
- [x] You signed the [Neo4j CLA](https://neo4j.com/developer/cla/#sign-cla) (Contributor License Agreement) so that we are allowed to ship your code in our library
- [x] Your contribution is covered by tests

### CODE REVIEW

**Comment by FlorentinD** [comment]
[[🍓] Check client for run_cypher usage and compatiblity with sessions](https://trello.com/c/zpxJR9LX/9630-%F0%9F%8D%93-check-client-for-runcypher-usage-and-compatiblity-with-sessions)

**Review by Mats-SX — COMMENTED** [review]
Here's half a review. Apologies for not finishing it.

File: graphdatascience/model/model.py [hunk 1]
```diff
@@ -190,27 +193,29 @@ def drop(self, failIfMissing: bool = False) -> "Series[Any]":
             The result of the drop operation.
 
         """
+        params = CallParameters(model_name=self._name, fail_if_missing=failIfMissing)
         if self._server_version < ServerVersion(2, 5, 0):
```
[the remark this answers is not in the document]
Comment (FlorentinD) [reply graphdatascience/model/model.py:197]: we should drop the support in a major version of the client. Will keep it for this PR

File: graphdatascience/model/model.py [hunk 2]
```diff
@@ -41,14 +43,15 @@ def _list_info(self) -> DataFrame:
         params = {"name": self.name()}
 
         # FIXME use call procedure + do post processing on the client side
+        # TODO really fixme
```
[the remark this answers is not in the document]
Comment (FlorentinD) [reply graphdatascience/model/model.py:46]: nope, I wanted to remove both in the end of this PR :D

**Pull request merged**